aspnetcore/razor-components/forms-validation.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Razor Components forms and validation
+author: guardrex
+description: Learn how to use forms and field validation scenarios in Razor Components.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/01/2019
+uid: razor-components/forms-validation
+---
+# Razor Components forms and validation
+
+By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.com/guardrex)
+
+Forms and validation are supported in Razor Components using [data annotations](xref:mvc/models/validation).
+
+> [!NOTE]
+> Forms and validation scenarios are likely to change with each preview release.
+
+The following `ExampleModel` type defines validation logic using data annotations:
+
+```csharp
+using System.ComponentModel.DataAnnotations;
+
+public class ExampleModel
+{
+    [Required]
+    [StringLength(10, ErrorMessage = "Name is too long.")]
```

Comment:
Please explain how localization with client side Blazor or a shared lib could be done.

Reply:
@seriouz This PR has already merged. It's best to open new issues using the **This Page** feedback button at the bottom of the topic. No worries on your question tho ... I can answer that here.

*It's in progress now.* See :point_right: https://github.com/aspnet/AspNetCore/issues/9386 ... It's due (tentatively) for Preview 8. We'll get it documented for the Preview 8 release. In the meantime, if you have design comments or questions, you can ask the engineers on that issue.